code/game/objects/items.dm

```diff
@@ -122,6 +122,64 @@ var/global/image/fire_overlay = image("icon" = 'icons/effects/fire.dmi', "icon_s
 			slot_drone_storage\
 		)
 
+
+	var/list/chameleon_blacklist = list()
+	var/list/chameleon_list = list()
+	var/chameleon_type = null
+	var/chameleon_name = "Item"
+	var/datum/action/item_action/chameleon/change/chameleon_change_action
+	var/chameleon_action_type = /datum/action/item_action/chameleon/change
+
+/datum/action/item_action/chameleon/change
+	name = "Chameleon Change"
+
+/datum/action/item_action/chameleon/change/Trigger()
+	if(!Checks())
+		return
+
+	var/obj/item/I = target
+	I.chameleon_change(owner)
+	return 1
+
+/obj/item/New()
+	..()
+	if(chameleon_type)
+		chameleon_change_action = new chameleon_action_type(src)
+		chameleon_change_action.name = "Change [chameleon_name] Appearance"
+		chameleon_blacklist += type
+		var/list/temp_list = typesof(chameleon_type)
+		for(var/U in temp_list - (chameleon_blacklist))
+			var/obj/item/V = new U

```

Comment:
you're actually creating an item of each type in nullspace just to put in a list?